src/main/java/com/powsybl/openloadflow/util/BusDistance.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.util;
+
+import com.powsybl.openloadflow.network.LfBus;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Sylvestre Prabakaran {@literal <sylvestre.prabakaran at rte-france.com>}
+ */
+public final class BusDistance {
+
+    private BusDistance() {
+    }
+
+    /**
+     * Breadth first search algorithm to compute distance between two LfBus in a network
+     * @param bus1              first LfBus (from which the breadth first search starts)
+     * @param bus2              second LfBus
+     * @param maxDistanceSearch the algorithm searches until this range and stops after this limit (or if every bus have been checked)
+     * @return                  measured distance (number of branches) or Integer.MAX_VALUE if bus2 is not found
+     */
+    public static int distanceBetweenBuses(LfBus bus1, LfBus bus2, int maxDistanceSearch) {
```

Comment:
This is an unusual way to implement a BFS. We can do it much simpler without these kind of set manipulation (busesToCheck.removeAll(checkedBuses)). Also we should take care for a BFS to have an orderer neighbors traversal (be careful with HashSets)

Reply:
Can you add a unit test for the method with various cases (exected distance, farther than maxDistance  etc..)

Reply:
Ok